src/GeneratedPlaylists/GeneratedPlaylist.tsx

```diff
@@ -1,17 +1,45 @@
-import React from 'react'
-
+import React, { Component } from 'react'
+import { CleanedAlbumTrack } from '../utils'
 import { getPlaylist } from '../apiCalls'
 
-const getSongsByGenre = () => {
-  // getPlaylist('hip+hop').then(data => console.log(data));
-}
+interface IProps {}
+interface IState { songs: CleanedAlbumTrack[],  selectedGenre: string }
+
+class GeneratedPlaylist extends Component<IProps, IState> {
+  constructor(props: any) {
+    super(props)
+    this.state = {
+      songs: [],
+      selectedGenre: ""
+    }
+  }
+
+  componentDidMount = () => {
+    getPlaylist(this.state.selectedGenre)
+    .then(data => this.setState({ songs: data }));
+  }
+
+  displaySongs = () => {
+    return this.state.songs.reduce((finalSongs: JSX.Element[], currentSong, index: number): JSX.Element[] => {
+      if (index < 10) {
+        const songToDisplay = 
+          <section key={index}>
+            <h3>{currentSong.songName}</h3>
+          </section>;
+        finalSongs.push(songToDisplay)
+      }
+      return finalSongs
+    }, [])
+  }
 
-function GeneratedPlaylist() {
-  console.log(getSongsByGenre());
-  return (
-    <section className='generated-playlist'>
-    </section>
-  )
+  render() {
+    const songsToDisplay = this.displaySongs()
+    return (
+      <section className='generated-playlist'>
+        {songsToDisplay}
```

Comment:
I think we could call `this.displaySongs()` directly inside of the section on line 39 instead of storing it inside of const beforehand.